File: test/corpus/basic/02-palette-generation/case.ts
import * as GoStyle from '../../../../src/go-style';

// Test fast palette generation (deterministic)
const colors = GoStyle.FastWarmPalette(3);
console.log(`FastWarmPalette count: ${colors.length}`);
for (let i = 0; i < colors.length; i++) {
  const c = colors[i];
  console.log(
    `Color ${i}: R=${c.R.toFixed(6)} G=${c.G.toFixed(6)} B=${c.B.toFixed(6)} Hex=${c.Hex()}`
  );
}

// Test fast happy palette
const happyColors = GoStyle.FastHappyPalette(2);
console.log(`FastHappyPalette count: ${happyColors.length}`);
for (let i = 0; i < happyColors.length; i++) {
  const c = happyColors[i];
  console.log(
    `Happy ${i}: R=${c.R.toFixed(6)} G=${c.G.toFixed(6)} B=${c.B.toFixed(6)} Hex=${c.Hex()}`
  );
}

// Test color creation from different spaces
const hslColor = GoStyle.Hsl(120.0, 1.0, 0.5); // Green from HSL
console.log(
  `HSL Color: R=${hslColor.R.toFixed(6)} G=${hslColor.G.toFixed(6)} B=${hslColor.B.toFixed(6)} Hex=${hslColor.Hex()}`
);

const hsvColor = GoStyle.Hsv(240.0, 1.0, 1.0); // Blue from HSV
console.log(
  `HSV Color: R=${hsvColor.R.toFixed(6)} G=${hsvColor.G.toFixed(6)} B=${hsvColor.B.toFixed(6)} Hex=${hsvColor.Hex()}`
);

File: test/corpus/basic/02-palette-generation/case.go
package main

import (
	"fmt"
	colorful "github.com/lucasb-eyer/go-colorful"
)

func main() {
	// Test fast palette generation (deterministic)
	colors := colorful.FastWarmPalette(3)
	fmt.Printf("FastWarmPalette count: %d\n", len(colors))
	for i, c := range colors {
		fmt.Printf("Color %d: R=%.6f G=%.6f B=%.6f Hex=%s\n", i, c.R, c.G, c.B, c.Hex())
	}
	
	// Test fast happy palette
	happyColors := colorful.FastHappyPalette(2)
	fmt.Printf("FastHappyPalette count: %d\n", len(happyColors))
	for i, c := range happyColors {
		fmt.Printf("Happy %d: R=%.6f G=%.6f B=%.6f Hex=%s\n", i, c.R, c.G, c.B, c.Hex())
	}
	
	// Test color creation from different spaces
	hslColor := colorful.Hsl(120.0, 1.0, 0.5) // Green from HSL
	fmt.Printf("HSL Color: R=%.6f G=%.6f B=%.6f Hex=%s\n", hslColor.R, hslColor.G, hslColor.B, hslColor.Hex())
	
	hsvColor := colorful.Hsv(240.0, 1.0, 1.0) // Blue from HSV  
	fmt.Printf("HSV Color: R=%.6f G=%.6f B=%.6f Hex=%s\n", hsvColor.R, hsvColor.G, hsvColor.B, hsvColor.Hex())
}